src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitArray = sourceString.trim().split(';');
+  const object = {};
+
+  for (const item of splitArray) {
+    if (item.trim().length > 0) {
+      const resultItemSplit = item.split(':');
```

Comment:
The 'split' operation should be followed by a check to ensure that 'resultItemSplit' has at least two elements before accessing them by index. This guards against possible errors if the input string is malformed.